checkers/rules/rules.go

```diff
@@ -785,3 +785,27 @@ func uncheckedInlineErr(m dsl.Matcher) {
 		Report("$err error is unchecked, maybe intended to check it instead of $err2").
 		At(m["err"])
 }
+
+//doc:summary Detects unsupported test and benchmark funcs
+//doc:tags    diagnostic experimental
+//doc:before  func TessstUnit(t *testing.T)
+//doc:after   func TestUnit(t *testing.T)
+func sloppyTestFuncName(m dsl.Matcher) {
+	m.Match(`func $test($_ *testing.T) { $*_ }`).
+		Where(!m["test"].Text.Matches("Test.*") &&
+			!m["test"].Text.Matches("test.*")).
+		Report("function $test should be of form TestXXX(t *testing.T)")
+
+	m.Match(`func $bench($_ *testing.B) { $*_ }`).
+		Where(!m["bench"].Text.Matches("Benchmark.*") &&
+			!m["bench"].Text.Matches("bench.*")).
+		Report("function $bench should be of form BenchmarkXXX(b *testing.B)")
+
+	m.Match(`func $test($_ *testing.T) { $*_ }`).
+		Where(m["test"].Text.Matches("test.*")).
```

Comment:
Doesn't it need anchors like `^test.*$`?
Because otherwise I think it will match things like `paratestBlah`.

Reply:
Thanks, fixed!